src/main/java/org/openrewrite/java/security/PartialPathTraversalVulnerability.java

```diff
@@ -190,8 +246,16 @@ private static boolean isSafePartialPathExpression(Expression expression) {
                     // CASE: ...startsWith(... + ...);
                     Expression right = unwrap(concatArgument.getRight());
                     if (right instanceof J.FieldAccess) {
-                        // CASE: ...startsWith(... + field);
-                        return true;
+                        // CASE:
+                        // - ...startsWith(... + File.separator);
+                        // - ...startsWith(... + File.seperatorChar);
+                        J.FieldAccess fieldAccess = (J.FieldAccess) right;
+                        String fieldName = fieldAccess.getName().getSimpleName();
+                        // TODO: Include filtering for declaring type being `java.io.File`.
+                        // This is currently not possible because `J.FieldAccess` does not have `JavaType.Variable`
+                        // information attached to it currently.
+                        // See https://github.com/openrewrite/rewrite/issues/1813
+                        return fieldName.equals("separatorChar") || fieldName.equals("separator");
                     } else if (right instanceof J.Literal) {
```

Comment:
thiss will handle the case of a static `java.io.File.separatorChar` import
```suggestion
                    } else if (right instanceof J.Identifier) {
                        J.Identifier ident = (J.Identifier)right;
                        return ident.getFieldType() != null && "separatorChar".equals(ident.getSimpleName())
                                && TypeUtils.isOfClassType(ident.getFieldType().getOwner(), "java.io.File");
                    } else if (right instanceof J.Literal) {
```